### ISSUE TITLE
Missing observable operators

### ISSUE THREAD

**Issue by rudiedev6** [qjon/angular2-filemanager#9]
I'm getting a lot of errors that looks similar to this:
"node_modules/@rign/angular2-filemanager/src/store/fileManagerApi.service.ts (32,23): Property 'of' does not exist on type 'typeof Observable'."

I'm including this using Angular CLI (Angular 4). Anyone else getting these errors?

**Comment by qjon** [comment]
It is strange. For me everything works fine, also all Unit test and builds on Travis works correctly, so for me it's hard to say what is wrong. Could you give more details (logs)? How looks your package.json? Maybe you can give me access to your project (if it is possible) then I can look on it.

**Comment by bargenson** [comment]
Same thing here.
The complete error message:
```
ERROR in /Users/bargenson/Development/fireloop-base-project/webapp/node_modules/@rign/angular2-filemanager/src/services/imageDataConverter.service.ts (42,8): Property 'concatMap' does not exist on type 'Observable<string>'.
/Users/bargenson/Development/fireloop-base-project/webapp/node_modules/@rign/angular2-filemanager/src/services/imageDataConverter.service.ts (48,29): Property 'of' does not exist on type 'typeof Observable'.
/Users/bargenson/Development/fireloop-base-project/webapp/node_modules/@rign/angular2-filemanager/src/services/imageDataConverter.service.ts (70,23): Property 'fromEvent' does not exist on type 'typeof Observable'.
/Users/bargenson/Development/fireloop-base-project/webapp/node_modules/@rign/angular2-filemanager/src/services/imageDataConverter.service.ts (87,34): Property 'fromEvent' does not exist on type 'typeof Observable'.

ERROR in /Users/bargenson/Development/fireloop-base-project/webapp/node_modules/@rign/angular2-filemanager/src/store/fileManagerApi.service.ts (27,23): Property 'of' does not exist on type 'typeof Observable'.
/Users/bargenson/Development/fireloop-base-project/webapp/node_modules/@rign/angular2-filemanager/src/store/fileManagerApi.service.ts (37,25): Property 'of' does not exist on type 'typeof Observable'.
/Users/bargenson/Development/fireloop-base-project/webapp/node_modules/@rign/angular2-filemanager/src/store/fileManagerApi.service.ts (39,25): Property 'empty' does not exist on type 'typeof Observable'.
/Users/bargenson/Development/fireloop-base-project/webapp/node_modules/@rign/angular2-filemanager/src/store/fileManagerApi.service.ts (53,25): Property 'of' does not exist on type 'typeof Observable'.
/Users/bargenson/Development/fireloop-base-project/webapp/node_modules/@rign/angular2-filemanager/src/store/fileManagerApi.service.ts (55,25): Property 'empty' does not exist on type 'typeof Observable'.
/Users/bargenson/Development/fireloop-base-project/webapp/node_modules/@rign/angular2-filemanager/src/store/fileManagerApi.service.ts (66,25): Property 'of' does not exist on type 'typeof Observable'.
/Users/bargenson/Development/fireloop-base-project/webapp/node_modules/@rign/angular2-filemanager/src/store/fileManagerApi.service.ts (68,25): Property 'empty' does not exist on type 'typeof Observable'.
/Users/bargenson/Development/fireloop-base-project/webapp/node_modules/@rign/angular2-filemanager/src/store/fileManagerApi.service.ts (83,25): Property 'of' does not exist on type 'typeof Observable'.
/Users/bargenson/Development/fireloop-base-project/webapp/node_modules/@rign/angular2-filemanager/src/store/fileManagerApi.service.ts (119,23): Property 'of' does not exist on type 'typeof Observable'.
/Users/bargenson/Development/fireloop-base-project/webapp/node_modules/@rign/angular2-filemanager/src/store/fileManagerApi.service.ts (126,25): Property 'of' does not exist on type 'typeof Observable'.
/Users/bargenson/Development/fireloop-base-project/webapp/node_modules/@rign/angular2-filemanager/src/store/fileManagerApi.service.ts (132,23): Property 'of' does not exist on type 'typeof Observable'.
/Users/bargenson/Development/fireloop-base-project/webapp/node_modules/@rign/angular2-filemanager/src/store/fileManagerApi.service.ts (148,23): Property 'of' does not exist on type 'typeof Observable'.
/Users/bargenson/Development/fireloop-base-project/webapp/node_modules/@rign/angular2-filemanager/src/store/fileManagerApi.service.ts (156,25): Property 'of' does not exist on type 'typeof Observable'.
/Users/bargenson/Development/fireloop-base-project/webapp/node_modules/@rign/angular2-filemanager/src/store/fileManagerApi.service.ts (195,25): Property 'of' does not exist on type 'typeof Observable'.

ERROR in /Users/bargenson/Development/fireloop-base-project/webapp/node_modules/@rign/angular2-filemanager/src/store/fileManagerBackendApi.service.ts (10,14): Class 'FileManagerBackendApiService' incorrectly implements interface 'IFileManagerApi'.
  Types of property 'removeSelectedFiles' are incompatible.
    Type '(selectedFiles: string[]) => Observable<boolean>' is not assignable to type '(selectedFiles: string[]) => Observable<boolean>'. Two different types with this name exist, but they are unrelated.
      Type 'Observable<boolean>' is not assignable to type 'Observable<boolean>'. Two different types with this name exist, but they are unrelated.
        Property 'source' is protected but type 'Observable<T>' is not a class derived from 'Observable<T>'.
/Users/bargenson/Development/fireloop-base-project/webapp/node_modules/@rign/angular2-filemanager/src/store/fileManagerBackendApi.service.ts (31,5): Type 'Observable<IOuterNode[]>' is not assignable to type 'Observable<IOuterNode[]>'. Two different types with this name exist, but they are unrelated.
  Property 'source' is protected but type 'Observable<T>' is not a class derived from 'Observable<T>'.
/Users/bargenson/Development/fireloop-base-project/webapp/node_modules/@rign/angular2-filemanager/src/store/fileManagerBackendApi.service.ts (62,5): Type 'Observable<IOuterNode>' is not assignable to type 'Observable<IOuterNode>'. Two different types with this name exist, but they are unrelated.
  Property 'source' is protected but type 'Observable<T>' is not a class derived from 'Observable<T>'.
/Users/bargenson/Development/fireloop-base-project/webapp/node_modules/@rign/angular2-filemanager/src/store/fileManagerBackendApi.service.ts (86,5): Type 'Observable<IOuterNode>' is not assignable to type 'Observable<IOuterNode>'. Two different types with this name exist, but they are unrelated.
/Users/bargenson/Development/fireloop-base-project/webapp/node_modules/@rign/angular2-filemanager/src/store/fileManagerBackendApi.service.ts (105,5): Type 'Observable<any>' is not assignable to type 'Observable<IOuterNode>'.
  Property 'source' is protected but type 'Observable<T>' is not a class derived from 'Observable<T>'.
/Users/bargenson/Development/fireloop-base-project/webapp/node_modules/@rign/angular2-filemanager/src/store/fileManagerBackendApi.service.ts (128,7): Type 'Observable<any>' is not assignable to type 'Observable<IOuterNode>'.
/Users/bargenson/Development/fireloop-base-project/webapp/node_modules/@rign/angular2-filemanager/src/store/fileManagerBackendApi.service.ts (148,5): Type 'Observable<any>' is not assignable to type 'Observable<IOuterFile>'.
  Property 'source' is protected but type 'Observable<T>' is not a class derived from 'Observable<T>'.
/Users/bargenson/Development/fireloop-base-project/webapp/node_modules/@rign/angular2-filemanager/src/store/fileManagerBackendApi.service.ts (165,5): Type 'Observable<IOuterFile[]>' is not assignable to type 'Observable<IOuterFile[]>'. Two different types with this name exist, but they are unrelated.
  Property 'source' is protected but type 'Observable<T>' is not a class derived from 'Observable<T>'.
/Users/bargenson/Development/fireloop-base-project/webapp/node_modules/@rign/angular2-filemanager/src/store/fileManagerBackendApi.service.ts (186,25): Property 'of' does not exist on type 'typeof Observable'.
/Users/bargenson/Development/fireloop-base-project/webapp/node_modules/@rign/angular2-filemanager/src/store/fileManagerBackendApi.service.ts (192,5): Type 'Observable<boolean>' is not assignable to type 'Observable<boolean>'. Two different types with this name exist, but they are unrelated.
/Users/bargenson/Development/fireloop-base-project/webapp/node_modules/@rign/angular2-filemanager/src/store/fileManagerBackendApi.service.ts (227,23): Property 'of' does not exist on type 'typeof Observable'.
/Users/bargenson/Development/fireloop-base-project/webapp/node_modules/@rign/angular2-filemanager/src/store/fileManagerBackendApi.service.ts (238,5): Type 'Observable<any>' is not assignable to type 'Observable<IOuterFile[]>'.
  Property 'source' is protected but type 'Observable<T>' is not a class derived from 'Observable<T>'.

ERROR in /Users/bargenson/Development/fireloop-base-project/webapp/node_modules/@rign/angular2-filemanager/src/store/fileManagerEffects.service.ts (19,8): Property 'map' does not exist on type 'Observable<IOuterFile[]>'.
/Users/bargenson/Development/fireloop-base-project/webapp/node_modules/@rign/angular2-filemanager/src/store/fileManagerEffects.service.ts (24,27): Property 'of' does not exist on type 'typeof Observable'.
/Users/bargenson/Development/fireloop-base-project/webapp/node_modules/@rign/angular2-filemanager/src/store/fileManagerEffects.service.ts (32,8): Property 'map' does not exist on type 'Observable<IOuterFile>'.
/Users/bargenson/Development/fireloop-base-project/webapp/node_modules/@rign/angular2-filemanager/src/store/fileManagerEffects.service.ts (40,31): Property 'of' does not exist on type 'typeof Observable'.
/Users/bargenson/Development/fireloop-base-project/webapp/node_modules/@rign/angular2-filemanager/src/store/fileManagerEffects.service.ts (47,8): Property 'map' does not exist on type 'Observable<boolean>'.
/Users/bargenson/Development/fireloop-base-project/webapp/node_modules/@rign/angular2-filemanager/src/store/fileManagerEffects.service.ts (50,31): Property 'of' does not exist on type 'typeof Observable'.
/Users/bargenson/Development/fireloop-base-project/webapp/node_modules/@rign/angular2-filemanager/src/store/fileManagerEffects.service.ts (57,8): Property 'map' does not exist on type 'Observable<boolean>'.
/Users/bargenson/Development/fireloop-base-project/webapp/node_modules/@rign/angular2-filemanager/src/store/fileManagerEffects.service.ts (60,31): Property 'of' does not exist on type 'typeof Observable'.
/Users/bargenson/Development/fireloop-base-project/webapp/node_modules/@rign/angular2-filemanager/src/store/fileManagerEffects.service.ts (68,8): Property 'map' does not exist on type 'Observable<IOuterFile>'.
/Users/bargenson/Development/fireloop-base-project/webapp/node_modules/@rign/angular2-filemanager/src/store/fileManagerEffects.service.ts (72,27): Property 'empty' does not exist on type 'typeof Observable'.
/Users/bargenson/Development/fireloop-base-project/webapp/node_modules/@rign/angular2-filemanager/src/store/fileManagerEffects.service.ts (83,8): Property 'map' does not exist on type 'Observable<IOuterFile[]>'.
/Users/bargenson/Development/fireloop-base-project/webapp/node_modules/@rign/angular2-filemanager/src/store/fileManagerEffects.service.ts (88,27): Property 'of' does not exist on type 'typeof Observable'.

ERROR in /Users/bargenson/Development/fireloop-base-project/webapp/node_modules/@rign/angular2-filemanager/src/services/currentDirectoryFiles.service.ts (55,5): Type 'Observable<storeEntities>' is not assignable to type 'Observable<storeEntities>'. Two different types with this name exist, but they are unrelated.
  Property 'source' is protected but type 'Observable<T>' is not a class derived from 'Observable<T>'.
/Users/bargenson/Development/fireloop-base-project/webapp/node_modules/@rign/angular2-filemanager/src/services/currentDirectoryFiles.service.ts (59,5): Type 'Observable<string[]>' is not assignable to type 'Observable<string[]>'. Two different types with this name exist, but they are unrelated.
  Property 'source' is protected but type 'Observable<T>' is not a class derived from 'Observable<T>'.
/Users/bargenson/Development/fireloop-base-project/webapp/node_modules/@rign/angular2-filemanager/src/services/currentDirectoryFiles.service.ts (63,5): Type 'Observable<string[]>' is not assignable to type 'Observable<string[]>'. Two different types with this name exist, but they are unrelated.
/Users/bargenson/Development/fireloop-base-project/webapp/node_modules/@rign/angular2-filemanager/src/services/currentDirectoryFiles.service.ts (76,8): Property 'withLatestFrom' does not exist on type 'Observable<string[]>'.
/Users/bargenson/Development/fireloop-base-project/webapp/node_modules/@rign/angular2-filemanager/src/services/currentDirectoryFiles.service.ts (97,23): Property 'combineLatest' does not exist on type 'typeof Observable'.

ERROR in /Users/bargenson/Development/fireloop-base-project/webapp/node_modules/@rign/angular2-filemanager/src/filemanager.component.ts (125,49): Argument of type 'FileManagerApiService' is not assignable to parameter of type 'INodeService'.
  Types of property 'load' are incompatible.
    Type '(nodeId?: string) => Observable<IOuterNode[]>' is not assignable to type '(nodeId: string) => Observable<IOuterNode[]>'.
      Type 'Observable<IOuterNode[]>' is not assignable to type 'Observable<IOuterNode[]>'. Two different types with this name exist, but they are unrelated.
        Property 'source' is protected but type 'Observable<T>' is not a class derived from 'Observable<T>'.
/Users/bargenson/Development/fireloop-base-project/webapp/node_modules/@rign/angular2-filemanager/src/filemanager.component.ts (129,5): Type 'Observable<ITreeData>' is not assignable to type 'Observable<ITreeData>'. Two different types with this name exist, but they are unrelated.
  Property 'source' is protected but type 'Observable<T>' is not a class derived from 'Observable<T>'.
/Users/bargenson/Development/fireloop-base-project/webapp/node_modules/@rign/angular2-filemanager/src/filemanager.component.ts (136,36): Argument of type 'Observable<ITreeData>' is not assignable to parameter of type 'Observable<ITreeData>'.
  Property 'source' is protected but type 'Observable<T>' is not a class derived from 'Observable<T>'.

ERROR in /Users/bargenson/Development/fireloop-base-project/webapp/node_modules/@rign/angular2-filemanager/src/toolbar/selectionDropDown/selection.component.ts (72,45): Property 'combineLatest' does not exist on type 'typeof Observable'.

ERROR in /Users/bargenson/Development/fireloop-base-project/webapp/node_modules/@rign/angular2-filemanager/src/filemanager.module.ts (23,35): Cannot find module '@ngrx/store-devtools'.

ERROR in Error encountered resolving symbol values statically. Could not resolve @ngrx/store-devtools relative to /Users/bargenson/Development/fireloop-base-project/webapp/node_modules/@rign/angular2-filemanager/src/filemanager.module.ts., resolving symbol FileManagerModule in /Users/bargenson/Development/fireloop-base-project/webapp/node_modules/@rign/angular2-filemanager/src/filemanager.module.ts, resolving symbol FileManagerModule in /Users/bargenson/Development/fireloop-base-project/webapp/node_modules/@rign/angular2-filemanager/src/filemanager.module.ts, resolving symbol FileManagerModule in /Users/bargenson/Development/fireloop-base-project/webapp/node_modules/@rign/angular2-filemanager/src/filemanager.module.ts

ERROR in ./~/@rign/angular2-filemanager/src/filemanager.module.ts
Module not found: Error: Can't resolve '@ngrx/store-devtools' in '/Users/bargenson/Development/fireloop-base-project/webapp/node_modules/@rign/angular2-filemanager/src'
 @ ./~/@rign/angular2-filemanager/src/filemanager.module.ts 32:0-59
 @ ./~/@rign/angular2-filemanager/main.ts
 @ ./src/app/app.module.ts
 @ ./src/main.ts
 @ multi webpack-dev-server/client?http://localhost:4200 ./src/main.ts
```

**Comment by bargenson** [comment]
The problem is because the version of RxJs you include with your package and the version I include myself are different. Could you distribute your package without RxJs included ?

**Comment by qjon** [comment]
I think that it is impossible to remove RxJs dependency. I think you should update yours dependency to the new one that was released. But from the other side, currently I'm working on some changes for new release and I almost finished it. They are available on branch feature/v1.2.0. You can use it in your project by changin one line in your package.json: 
`"@rign/angular2-filemanager": "https://github.com/qjon/angular2-filemanager.git#feature/v1.2.0"`

If you do that, please give me feedback if it works, then I update master branch and make a new release.
